MusicTaste/MusicTaste/View Controllers/Login/ConnectButton.m

```diff
@@ -0,0 +1,48 @@
+//
+//  ConnectButton.m
+//  MusicTaste
+//
+//  Created by Aman Abraham on 7/8/22.
+//
+
+#import "ConnectButton.h"
+
+@implementation ConnectButton
+
++ (instancetype)buttonWithType:(UIButtonType)buttonType
+{
+    ConnectButton *button = [super buttonWithType:buttonType];
+    button.backgroundColor = [self customBackgroundColor];
+    button.contentEdgeInsets = UIEdgeInsetsMake(11.75, 32.0, 11.75, 32.0);
+    button.layer.cornerRadius = 20.0;
+    button.translatesAutoresizingMaskIntoConstraints = NO;
+
+    return button;
+}
+
++ (UIColor *)customBackgroundColor
+{
+    static UIColor *color;
+
+    if (!color)
+        color = [UIColor colorWithRed:(29.0 / 255.0) green:(185.0 / 255.0) blue:(84.0 / 255.0) alpha:1.0];
+
+    return color;
+}
+
++ (UIColor *)customHighlightedBackgroundColor
+{
+    static UIColor *color;
+
+    if (!color)
+        color = [UIColor colorWithRed:(24.0 / 255.0) green:(172.0 / 255.0) blue:(77.0 / 255.0) alpha:1.0];
+
+    return color;
+}
+
+- (void)setHighlighted:(BOOL)highlighted
+{
+    self.backgroundColor = highlighted ? [self.class customHighlightedBackgroundColor] : [self.class customBackgroundColor];
+}
+
```

Comment:
Nice example of creating UI with code. This would be a great example to understand deeply if you can.

By the way, side note: Writing code is my preferred way to do UI; I don't like storyboards as I find them very finicky. Also, code review with storyboards is hard.